lib/beacon/content.ex

```diff
@@ -367,17 +398,17 @@ defmodule Beacon.Content do
   @doc type: :pages
   @spec create_page(map()) :: {:ok, Page.t()} | {:error, Ecto.Changeset.t()}
   def create_page(attrs) when is_map(attrs) do
-    create = fn attrs ->
-      %Page{}
-      |> Page.create_changeset(attrs)
-      |> Repo.insert()
-    end
-
     Repo.transact(fn ->
-      with {:ok, page} <- create.(attrs),
+      changeset = Page.create_changeset(%Page{}, attrs)
+
+      with %Changeset{valid?: true} <- validate_page_template(changeset),
+           {:ok, page} <- Repo.insert(changeset),
            {:ok, _event} <- create_page_event(page, "created"),
            %Page{} = page <- Lifecycle.Page.after_create_page(page) do
         {:ok, page}
+      else
+        %Changeset{} = changeset -> {:error, changeset}
+        error -> error
       end
```

Comment:
It would be nice if we did not need `else` here.  If we could do
```elixir
with {:ok, changeset} <- validate_page_template,
```
then the error case would already be tagged 🙂

Reply:
Yep good catch. I'll make the change thanks